lib/content-negotiation.js

```diff
@@ -43,15 +46,16 @@ const mediaTypeMarshallers = {
         marshall: (any) => Buffer.from(any),
     },
     "text/plain": {
-        unmarshall: (buffer) => textDecoder.decode(buffer),
+        unmarshall: (buffer, charset) => resolveDecoder(charset).decode(buffer),
         marshall: (string) => textEncoder.encode("" + string),
     },
     "application/json": {
         unmarshall: jsonUnmarshall,
         marshall: jsonMarshall,
     },
     "application/x-www-form-urlencoded": {
-        unmarshall: (buffer) => querystring.parse(textDecoder.decode(buffer)),
+        unmarshall: (buffer) =>
+            querystring.parse(utf8TextDecoder.decode(buffer)),
```

Comment:
should we also resolve the charset here?

Reply:
I checked [the spec](https://url.spec.whatwg.org/#urlencoded-parsing) beforehand, and we should assume UTF-8